### PR TITLE
<TableListItem/> - new component

### PR DESCRIPTION
**Pull request by Lukermelesh**
### 🔦 Summary

...

### ✅ Checklist

- [ ] 👨‍🎨 UX change is approved by the Design System UX 
- [ ] 👨‍💻 API change is approved by the UI Infra Developers 
- 📚 Change is documented
  - [ ] Story
  - [ ] API description
  - [ ] Cheatsheet
  - [ ] Other (explain)
- 🔬 Change is tested
  - [ ] Component
  - [ ] Visual test

### CODE REVIEW

**Comment by WixBuildServer** [comment]
### 🤖 Hi, I'm #Janet!

  A deploy preview of your PR was successfully released 🖼.

  You can find it under one of the following URLs:
  * https://www.wix.com/pages/wix-style-react.pr_5788/

File: src/TableListItem/TableListItem.st.css [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+  background-color: value(B60);
+}
+
+.optionsContainer {
+  display: grid;
+  width: 100%;
+}
+
+.align {
+  -st-states: left, center, right;
```
Comment (EgoziE) [inline src/TableListItem/TableListItem.st.css:53]: You can use an enum:
```suggestion
  -st-states: align(enum(left, center, right));
```

File: src/TableListItem/TableListItem.st.css [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+  margin-left: 6px;
+  margin-right: 18px;
+}
+
+.align:left {
```
Comment (EgoziE) [inline src/TableListItem/TableListItem.st.css:58]: And then:

```suggestion
.align:align(left) {
```

File: src/TableListItem/TableListItem.js [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+  className: PropTypes.string,
+
+  /**
+   width supports px/%/fr
+   */
+  options: PropTypes.arrayOf(
+    PropTypes.shape({
+      value: PropTypes.node.isRequired,
+      width: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
+      align: PropTypes.oneOf(Object.keys(ALIGN)),
```
Comment (EgoziE) [inline src/TableListItem/TableListItem.js:117]: We have an annoying bug in our docs generator, please write the values explicitly.

File: src/TableListItem/TableListItem.js [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+    }),
+  ).isRequired,
+
+  /**
+    Extra space on top and bottom of list item
+   */
+  verticalPadding: PropTypes.oneOf(Object.keys(VERTICAL_PADDING)),
```
Comment (EgoziE) [inline src/TableListItem/TableListItem.js:124]: We have an annoying bug in our docs generator, please write the values explicitly.

File: src/TableListItem/docs/index.story.js [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+                    { value: 'Investing', align: 'right' },
+                    { value: '4 posts', align: 'center', width: '80px' },
+                    { value: '30 April 2020', align: 'left' },
+                  ]}
+                />
+              </div>`,
+          }),
+
+          code({
+            title: 'Full Interactive Preview',
```
Comment (EgoziE) [inline src/TableListItem/docs/index.story.js:123]: No need for the same example in a different view, can be deleted

File: src/TableListItem/index.d.ts [hunk 6]
```diff
@@ -9,6 +9,7 @@ export type TableListItemColumn = {
 export interface TableListItemProps {
   dataHook?: string;
   className?: string;
+  onClick?: Function;
```
Comment (EgoziE) [inline src/TableListItem/index.d.ts:12]:
```suggestion
  onClick?: React.MouseEventHandler<HTMLButtonElement>;
```

File: src/TableListItem/TableListItem.js [hunk 7]
```diff
@@ -0,0 +1,181 @@
...
+  className: PropTypes.string,
+
+  /**
+   width supports px/%/fr
+   */
+  options: PropTypes.arrayOf(
+    PropTypes.shape({
+      value: PropTypes.node.isRequired,
+      width: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
+      align: PropTypes.oneOf([ALIGN.left, ALIGN.center, ALIGN.right]),
```
Comment (EgoziE) [inline src/TableListItem/TableListItem.js:120]:
```suggestion
      align: PropTypes.oneOf(['left', 'center', 'right']),
```

File: src/TableListItem/TableListItem.js [hunk 8]
```diff
@@ -0,0 +1,181 @@
...
+    }),
+  ).isRequired,
+
+  /**
+    Extra space on top and bottom of list item
+   */
+  verticalPadding: PropTypes.oneOf([
+    VERTICAL_PADDING.medium,
+    VERTICAL_PADDING.small,
+  ]),
```
Comment (EgoziE) [inline src/TableListItem/TableListItem.js:130]:
```suggestion
  verticalPadding: PropTypes.oneOf(['small', 'medium']),
```

File: src/TableListItem/TableListItem.js (hunk 7 above)
Comment (EgoziE) [reply src/TableListItem/TableListItem.js:120]: Sorry I wasn't clear enough, the values must be explicit.

File: src/TableListItem/TableListItem.js [hunk 9]
```diff
@@ -36,18 +36,21 @@ const TableListItem = ({
...
   dragDisabled,
   showDivider,
+  onClick,
   className,
   dataHook,
 }) => {
   const DragHandleIcon = dragDisabled ? DragHandleDisabled : DragHandle;
   return (
     <div
+      onClick={onClick}
```
Comment (EgoziE) [inline src/TableListItem/TableListItem.js:46]: Should the whole component be clickable? @justinara

File: src/TableListItem/TableListItem.js [hunk 10]
```diff
@@ -157,11 +163,16 @@ TableListItem.propTypes = {
...
+  /**
+    Called when the item is clicked
+   */
+  onClick: PropTypes.func,
 };
 
 TableListItem.defaultProps = {
   onCheckboxChange: () => {},
-  verticalPadding: VERTICAL_PADDING.small,
+  verticalPadding: VERTICAL_PADDING.medium,
```
Comment (EgoziE) [inline src/TableListItem/TableListItem.js:175]:
```suggestion
  verticalPadding: 'medium',
```

**Pull request merged**